packages/@aws-cdk/aws-dynamodb/lib/replica-provider.ts

```diff
@@ -27,10 +27,34 @@ export class ReplicaProvider extends NestedStack {
    */
   public static getOrCreate(scope: Construct, props: ReplicaProviderProps = {}) {
     const stack = Stack.of(scope);
+    this.checkManagedPoliciesLimit(stack);
     const uid = '@aws-cdk/aws-dynamodb.ReplicaProvider';
     return stack.node.tryFindChild(uid) as ReplicaProvider ?? new ReplicaProvider(stack, uid, props);
   }
 
+  public static clearGetOrCreateCalls() {
+    this.getOrCreateCalls.clear();
+  }
```

Comment:
I hate hate hate adding these methods that are only used in tests 😭.

Can we instead hack into the `getOrCreateCalls` field in the test with type casts?